frontend/public/components/resource-pages.ts

```diff
@@ -537,10 +537,11 @@ export const baseListPages = ImmutableMap<ResourceMapKey, ResourceMapValue>()
     ),
   );
 
-export const resourceListPages = ImmutableMap<ResourceMapKey, ResourceMapValue>()
-  .merge(baseListPages)
-  .withMutations((map) => {
-    plugins.registry.getResourceListPages().forEach((page) => {
-      addResourcePage(map, page);
+export const getResourceListPages = (pluginPages: ResourceListPage[]) =>
```

Comment:
For those callers that don't need to work with plugin contributed pages:

```suggestion
export const getResourceListPages = (pluginPages: ResourceListPage[] = []) =>
```